### PR TITLE
FIX: cypress recommendation for old stories

### PR DESCRIPTION
**Pull request by eagerterrier**
very old articles/stories should not display recommendations

### CODE REVIEW

**Comment by karinathomasbbc** [comment]
I have an idea for how to reduce duplication in the code - just working on it now and will push changes shortly. I would like to run the tests against the live environment

**Comment by eagerterrier** [comment]
> I have an idea for how to reduce duplication in the code - just working on it now and will push changes shortly. I would like to run the tests against the live environment

sorry - had to revert these. Cypress has to return a promise and chain things. So the keeping it in a function like that returns truthy regardless of whether operand was `>` or `<` 

returning the thing cypress thing itself (which is an object of some kind - probably a promisified Cypress object?) also didn't work.

The only other alternative would be to do this bit by plain JS. Not sure if that's possible or just an anti-pattern.

Going to revert this and make live (assuming tests pass)

File: package.json [hunk 1]
```diff
@@ -60,7 +62,7 @@
     "test:puppeteer": "jest --ci --env=jsdom --colors ./puppeteer",
     "test:unit": "test -z $CI && NODE_OPTIONS=--experimental-vm-modules jest --env=jsdom --coverage --colors --testPathIgnorePatterns=\"src/integration\" \"puppeteer\" \"ws-nextjs-app\"  || NODE_OPTIONS=--experimental-vm-modules jest --ci --runInBand --env=jsdom --coverage --colors --testPathIgnorePatterns=\"src/integration\" \"puppeteer\" \"ws-nextjs-app\"",
     "test:unit:watch": "yarn test:unit -- --watch",
-    "test:unit:updatesnapshots": "NODE_OPTIONS=--experimental-vm-modules jest --updateSnapshot --env=jsdom --coverage --colors --testPathIgnorePatterns=\"src/integration\" \"puppeteer\" \"ws-nextjs-app\"",
+    "test:unit:updatesnapshots": "NODE_OPTIONS=--experimental-vm-modules jest --updateSnapshot --env=jsdom --coverage --silent=false --colors --testPathIgnorePatterns=\"src/integration\" \"src/app/legacy\" \"src/app/pages\" \"src/app/lib\" \"src/app/contexts\" \"src/app/components\" \"puppeteer\" \"ws-nextjs-app\"",
```
Comment (andrewscfc) [inline package.json]: Why are these tests being ignore on snapshot update?

File: package.json [hunk 2]
```diff
@@ -36,7 +36,9 @@
     "build:test": "cp envConfig/test.env .env && NODE_ENV=production webpack",
     "build:test:debug": "rm -rf build && awk '{sub(/LOG_DIR=.+/,\"LOG_DIR='log'\")}1' envConfig/test.env > .env && NODE_ENV=production webpack",
     "cypress": "cypress run --config-file cypress-simorgh.json",
-    "cypress:onefile": "CYPRESS_APP_ENV=live CYPRESS_SMOKE=false cypress run --config-file cypress-simorgh.json --spec '/Applications/Code/simorgh/cypress/integration/pages/storyPage/*'",
+    "cypress:onefile": "CYPRESS_APP_ENV=live CYPRESS_SMOKE=false cypress run --config-file cypress-simorgh.json --spec './cypress/integration/pages/storyPage/*'",
+    "cypress:oneservice": "CYPRESS_APP_ENV=live CYPRESS_SMOKE=false CYPRESS_ONLY_SERVICE=mundo cypress run --config-file cypress-simorgh.json",
+    "cypress:oneserviceandonefile": "CYPRESS_APP_ENV=live CYPRESS_SMOKE=false CYPRESS_ONLY_SERVICE=mundo cypress run --config-file cypress-simorgh.json --spec './cypress/integration/pages/storyPage/*'",
```
Comment (andrewscfc) [inline package.json:41]: Is this list of script getting a bit big? Pretty harmless but just wondering if these were more intended for debugging?

**Comment by karinathomasbbc** [comment]
> > I have an idea for how to reduce duplication in the code - just working on it now and will push changes shortly. I would like to run the tests against the live environment
> 
> sorry - had to revert these. Cypress has to return a promise and chain things. So the keeping it in a function like that returns truthy regardless of whether operand was `>` or `<`
> 
> returning the thing cypress thing itself (which is an object of some kind - probably a promisified Cypress object?) also didn't work.
> 
> The only other alternative would be to do this bit by plain JS. Not sure if that's possible or just an anti-pattern.
> 
> Going to revert this and make live (assuming tests pass)

Right, I've tried again and ran the tests against test & live STY pages, and the recommendation tests seem to pass (or not run) as expected. 🤞

File: package.json (hunk 2 above)
Comment (eagerterrier) [reply package.json:41]: no! 

They were really to speed up certain testing and so knowledge isn't lost. 

if you just want to do one service - e.g. just test mundo - here's the command; if you want to do one folder of tests - here's the command, if you want to do one folder _and_ one service (fastest!) then do that. 

Helps devs run targeted tests quickly.

Maybe we should write this up or demo it?

File: package.json (hunk 1 above)
Comment (eagerterrier) [reply package.json]: oh yeah, good spot - this was to speed up a test and shouldn't have been committed!

**Review by amoore108 — APPROVED** [review]

**Pull request merged**